### ISSUE TITLE
Ground truth files?

### ISSUE THREAD

**Issue by venky4121994** [scumatteo/DeepVisualOdometry#1]
Hi can I know where to obtain the ground truth files for the sequences. When I try downloading it from KITT it is a .txt file.
If you have the .npy files can you provide it please?

**Comment by scumatteo** [comment]
> Hi can I know where to obtain the ground truth files for the sequences. When I try downloading it from KITT it is a .txt file. If you have the .npy files can you provide it please?

Hi, 
each file of the KITTI ground truth arranges the transfromation matrix T as a 3x4 matrix on a single line, so the first four element corresponds to the first row of the matrix, the second four to the second row and the third four to the third row.
These matrix corresponds to the pose (3x3 R matrix for the orientation and 3x1 vector for the translation) of the camera. 

In our project we needed the relative transformation between each consecutive pose. To obtain that we applied these formulas:
R_relative = inverse(R_1) @ R_2
t_relative = inverse(R_1) @ (t_2 - t_1)

Where R_2 and t_2 are the next rotation matrix and translation vector respectively and R_1, t_1 are the current ones.

Repeating this for each consecutive matrix and for each KITTI sequence we obtained the entire ground truth of relative transformations. 

Lastly, we saved these data as numpy files.

I attach a zip file that contains for each KITTI sequence four numpy files:

- (n_sequence)_path is the ground truth of the poses
- (n_sequence)_rt_euler is the file of the relative transformations that use euler angle as rotation encoding (roll-pitch-yaw order)
- (n_sequence)_rt_matrix is the file of the relative transformations that use rotation matrices as rotation encoding
- (n_sequence)_rt_quaternion is the file of the relative transformations that use quaternions as rotation encoding (xyzw order).

You're welcome
[numpy.zip](https://github.com/scumatteo/DeepVisualOdometry/files/8387304/numpy.zip)
